tensorflow/python/eager/function.py

```diff
@@ -105,9 +105,17 @@ def _make_input_signature_hashable(elem):
   Returns:
     A hashable object for the requested input signature
   """
+  try:
+    hash(elem)
+  except TypeError:
+    return _make_input_signature_hashable_helper(elem)
```

Comment:
nit: This is a very simple logic so I think we can skip splitting a helper.

Reply:
agree, python function call has 100us overhead. let us directly put this inside the code